src/Uno.UI/UI/Xaml/DependencyPropertyDetailsCollection.cs

```diff
@@ -22,15 +23,10 @@ partial class DependencyPropertyDetailsCollection : IDisposable
 		private DependencyPropertyDetails? _dataContextPropertyDetails;
 		private DependencyPropertyDetails? _templatedParentPropertyDetails;
 
-		private readonly static ArrayPool<short> _offsetsPool = ArrayPool<short>.Shared;
-		private readonly static LinearArrayPool<DependencyPropertyDetails?> _pool = LinearArrayPool<DependencyPropertyDetails?>.CreateAutomaticallyManaged(BucketSize, 16);
+		private DependencyPropertyDetails[] _entries = Array.Empty<DependencyPropertyDetails>();
```

Comment:
This change basically removes the pooling to allocate every time. This is curious that this leads to improvements.

Reply:
Pooling doesn't seem very much beneficial in this case, I think. Those arrays are more commonly long living, while pooling is more useful for short living objects.